### PR TITLE
feat: Added view timeout duration as a config parameter

### PR DESCRIPTION
**Pull request by brunoffranca**
## What ❔

The hardcoded 2s view timeout was causing problems in Era testnet because the block time there is longer (>5s). View timeout needs to be a configurable parameter since different chains will want different block times.

### CODE REVIEW

File: node/components/bft/src/chonky_bft/proposer.rs [hunk 1]
```diff
@@ -27,15 +26,15 @@ pub(crate) async fn run_proposer(
...
+            &ctx.with_timeout(cfg.view_timeout),
             cfg.clone(),
             justification,
         )
         .await
         {
             Ok(proposal) => proposal,
             Err(ctx::Error::Canceled(_)) => {
-                tracing::error!("run_proposer(): timed out while creating a proposal");
+                tracing::warn!("run_proposer(): timed out while creating a proposal");
```
Comment (pompon0) [inline node/components/bft/src/chonky_bft/proposer.rs:37]: maybe just info? Afaiu it is expected to have infrequent blocks on some chains.

File: node/components/executor/src/lib.rs [hunk 2]
```diff
@@ -39,6 +39,8 @@ pub struct Config {
     pub public_addr: net::Host,
     /// Maximal size of the block payload.
     pub max_payload_size: usize,
+    /// The duration of the view timeout, in milliseconds.
+    pub view_timeout: usize,
```
Comment (pompon0) [inline node/components/executor/src/lib.rs:43]: why removing strong typing?, use time::Duration

File: node/tools/src/proto/mod.proto [hunk 3]
```diff
@@ -98,6 +98,9 @@ message AppConfig {
   // Maximal size of the block payload.
   optional uint64 max_payload_size = 5; // required; bytes
 
+  // The duration of the view timeout.
+  optional uint64 view_timeout = 19; // required; milliseconds
```
Comment (pompon0) [inline node/tools/src/proto/mod.proto:102]: std.Duration

File: node/components/bft/src/chonky_bft/proposer.rs (hunk 1 above)
Comment (brunoffranca) [reply node/components/bft/src/chonky_bft/proposer.rs:37]: True, but maybe in that case the chain should be warned to increase the timeout. Overall, I don't think having a lot of timeouts should be considered normal operation.

File: node/components/executor/src/lib.rs (hunk 2 above)
Comment (brunoffranca) [reply node/components/executor/src/lib.rs:43]: I thought we didn't have Duration in Proto. I'll fix it.

**Review by pompon0 — DISMISSED** [review]

**Pull request merged**